### ISSUE TITLE
is it possible to create username and password for clients?

### ISSUE THREAD

**Issue by jbsaibes** [nopnop2002/esp-idf-mqtt-broker#49]
i was wondering if it is possible to assign a username/password for clients that would connect to the broker?

**Comment by nopnop2002** [comment]
This is a good idea.
Try latest.
![config-8](https://github.com/user-attachments/assets/d5b84006-1c70-4efa-89d2-c21241b79b5a)

**Comment by jbsaibes** [comment]
yes i just tried and it works thanks! next request :) two nice features to have :
1) having a list of clients ( cid, user and password) that broker can use for authentication
2) having list of topics that above clients can subscribe/publish to

**Comment by nopnop2002** [comment]
Are there any open source brokers with this kind of functionality?


If there is one, I would like to imitate its implementation and settings..

There are no plans to implement unique access restrictions specific to this broker.

**Comment by jbsaibes** [comment]
yes there is https://github.com/eclipse/mosquitto it one of the most popular mqtt brokers

**Comment by nopnop2002** [comment]
I'm using Mosquito, but I don't know the settings to achieve this.

In mosquitto, what settings do I need to do to achieve this?

```
having a list of clients ( cid, user and password) that broker can use for authentication
having list of topics that above clients can subscribe/publish to
```

**Comment by jbsaibes** [comment]
https://mosquitto.org/man/mosquitto-conf-5.html check general options>acl file option  and "password_file" option .there is a description there.

**Comment by nopnop2002** [comment]
Thank you for the information.
I'll try to investigate, but it will take some time.

If we use a password file, we will need a file system like SPIFFS.
